### PR TITLE
[Chart] Adding manila helm chart

### PR DESCRIPTION
**Pull request by stannum-l**
**What is the purpose of this pull request?:**
Adding manila service support for openstack-helm.

**What issue does this pull request address?**:
Issue#178 

**Notes for reviewers to consider:**
None

**Specific reviewers for pull request:**
@intlabs @alanmeadows @v1k0d3n @larryrensing

### CODE REVIEW

File: manila/templates/etc/_manila.conf.tpl [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+region_name = {{.Values.global.region}}
+username = {{ .Values.global.cinder_service_user }}
+password = {{ .Values.global.cinder_service_password }}
+user_domain_name = {{.Values.global.keystone_service_domain}}
+project_name = {{.Values.global.keystone_service_project}}
+project_domain_name = {{.Values.global.keystone_service_domain}}
+
+
+[nova]
+auth_url = {{.Values.global.keystone_api_endpoint_protocol_admin}}://{{include "keystone_api_endpoint_host_admin" .}}:{{ .Values.global.keystone_api_port_admin }}/v3
```
Comment (alanmeadows) [inline manila/templates/etc/_manila.conf.tpl]: Same as above - there is a pattern for endpoint lookup

File: manila/templates/etc/_manila.conf.tpl [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+auth_plugin = v3password
+region_name = {{.Values.global.region}}
+username = {{ .Values.global.nova_service_user }}
+password = {{ .Values.global.nova_service_password }}
+user_domain_name = {{.Values.global.keystone_service_domain}}
+project_name = {{.Values.global.keystone_service_project}}
+project_domain_name = {{.Values.global.keystone_service_domain}}
+
+[neutron]
+url = {{.Values.global.neutron_api_endpoint_protocol_internal}}://{{include "neutron_api_endpoint_host_internal" .}}:{{ .Values.global.neutron_api_port_internal }}
```
Comment (alanmeadows) [inline manila/templates/etc/_manila.conf.tpl]: Same

File: manila/templates/etc/_manila.conf.tpl [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+project_domain_name = {{.Values.global.keystone_service_domain}}
+
+[oslo_concurrency]
+lock_path = /var/lib/manila/tmp
+
+[database]
+connection = mysql+pymysql://{{ .Values.database.manila_user }}:{{ .Values.database.manila_password }}@{{ .Values.database.address }}/{{ .Values.database.manila_database_name }}
+
+[keystone_authtoken]
+auth_uri = {{ .Values.keystone.auth_uri }}
```
Comment (alanmeadows) [inline manila/templates/etc/_manila.conf.tpl]: Same

File: manila/templates/job-db-init.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+{{- $dependencies := .Values.dependencies.db_init }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: manila-db-init
+spec:
+  template:
+    metadata:
+      annotations:
+        pod.beta.kubernetes.io/init-containers:
```
Comment (alanmeadows) [inline manila/templates/job-db-init.yaml]: There is now a macro pattern to follow for dependency checking in common to condense this

File: manila/templates/etc/_manila.conf.tpl [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+rpc_response_timeout = 60
+rpc_workers = 1
+
+quota_share_networks = 0
+
+[cinder]
+auth_url = {{.Values.global.keystone_api_endpoint_protocol_admin}}://{{include "keystone_api_endpoint_host_admin" .}}:{{ .Values.global.keystone_api_port_admin }}/v3
+auth_plugin = v3password
+region_name = {{.Values.global.region}}
```
Comment (v1k0d3n) [inline manila/templates/etc/_manila.conf.tpl]: @stannum-l to @alanmeadows' point, this document could help some: [Openstack-Helm Philosophy](https://github.com/att-comdev/openstack-helm/blob/master/docs/helm_overrides.md#values-philosophy).

File: manila/templates/etc/_manila.conf.tpl [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+osapi_share_listen = 0.0.0.0
+
+rpc_response_timeout = 60
+rpc_workers = 1
+
+quota_share_networks = 0
+
+[cinder]
+auth_url = {{.Values.global.keystone_api_endpoint_protocol_admin}}://{{include "keystone_api_endpoint_host_admin" .}}:{{ .Values.global.keystone_api_port_admin }}/v3
```
Comment (v1k0d3n) [inline manila/templates/etc/_manila.conf.tpl]: @alanmeadows we should get @intlabs work pulled in, and like you recommended in #142 split some of the work if it eases his load as he transitions (i'm sure he'll be pretty busy or even offline soon). this way we can clearly define that this *is* the expectation going forward. once it's in, we can remove the development references around endpoint "dragons". :)

File: manila/templates/job-db-init.yaml (hunk 4 above)
Comment (stannum-l) [inline manila/templates/job-db-init.yaml]: Will fix in the next patch set.

File: manila/templates/etc/_manila.conf.tpl (hunk 5 above)
Comment (stannum-l) [inline manila/templates/etc/_manila.conf.tpl]: Thanks for the feedback.  I will do another pass to change all the auth_url's (and others) in the various service this week.  This was the template pulled from the SAP OS-helm to understand the dependencies needed by manila.

File: manila/templates/deployment-scheduler.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+{{ tuple $envAll $dependencies | include "helm-toolkit.kubernetes_entrypoint_init_container" | indent 10 }}
+        ]'
+    spec:
+      nodeSelector:
+        {{ .Values.labels.node_selector_key }}: {{ .Values.labels.node_selector_value }}
+      containers:
+        - name: manila-scheduler
+          image: {{ .Values.images.scheduler }}
+          imagePullPolicy: {{ .Values.images.pull_policy }}
+          command:
```
Comment (larryrensing) [inline manila/templates/deployment-scheduler.yaml]: Add some resource limits to this manifest

File: Makefile [hunk 8]
```diff
@@ -17,10 +17,10 @@
 B64_DIRS := helm-toolkit/secrets
 B64_EXCLUDE := $(wildcard helm-toolkit/secrets/*.b64)
 
-CHARTS := ceph mariadb etcd postgresql rabbitmq memcached keystone glance horizon neutron nova cinder heat maas
```
Comment (larryrensing) [inline Makefile]: Looks like the removal of MaaS and PostgreSQL put you into a merge conflict

**Comment by wilkers-steve** [comment]
Thanks for this work @stannum-l.  We'll file a blueprint once we're in OpenStack to track this work going forward.  Until then, we'll leave this PR open for you to reference after that point.

**Comment by v1k0d3n** [comment]
closing this PR in favor of moving the work to Openstack proper: https://github.com/openstack/openstack-helm

**Pull request closed without merging**